app/services/service_provider_request_handler.rb

```diff
@@ -10,9 +10,12 @@ def call
     pull_request_id_from_current_sp_session_id
 
     delete_sp_request_if_session_has_matching_request_id
-    ServiceProviderRequestProxy.create!(attributes)
 
-    StoreSpMetadataInSession.new(session: session, request_id: request_id).call
+    service_provider_request = ServiceProviderRequestProxy.create!(attributes)
```

Comment:
separate topic, but while we're here: I think that it's probably time to combine `ServiceProviderRequest` and `ServiceProviderRequestProxy`. The proxy is leftover from a migration from an AR model to a redis model, and now that we only have the redis one, we could probably simplify

Reply:
Agree, we can probably pull the classes together since `ServiceProviderRequest` is basically a struct at this point.